cmd/subnetcmd/join.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/spf13/cobra"
+)
+
+var avagoConfigPath string
+
+// avalanche subnet deploy
+func newJoinCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "join [subnetName]",
+		Short: "Instruct a validator node to begin validating a new subnet",
+		Long: `Instruct a validator node to begin validating a new subnet.
+The NodeID of that validator node must have been whitelisted by one of the 
+subnet's control keys for this to work.
+
+The node also needs to be restarted.
+If --avalanchego-config is provided, this command tries to edit the config file at that path
+(consider correct permissions).`,
+		RunE: joinCmd,
+		Args: cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&avagoConfigPath, "avalanchego-config", "", "file path of the avalanchego config file")
+	return cmd
+}
+
+func joinCmd(cmd *cobra.Command, args []string) error {
+	/*
+		chains, err := validateSubnetName(args)
+		if err != nil {
+			return err
+		}
+		subnetName := chains[0]
+
+		sc, err := app.LoadSidecar(subnetName)
+		if err != nil {
+			return err
+		}
+
+		subnetID := sc.SubnetID
+		if subnetID == ids.Empty {
+			return errNoSubnetID
+		}
+	*/
+	subnetIDstr := "29Zd5yhP7Yb2cTebBbUVKUjHHNviHzgj1y9kKJsvMn2dyWnkpG"
+	var network models.Network
+	networkStr, err := prompts.CaptureList(
+		"Choose a network to deploy on (this command only supports public networks)",
+		[]string{models.Fuji.String(), models.Mainnet.String()},
+	)
+	if err != nil {
+		return err
+	}
+	network = models.NetworkFromString(networkStr)
+	networkLower := strings.ToLower(network.String())
+
+	if avagoConfigPath == "" {
+		printJoinCmd(subnetIDstr, networkLower)
+	} else {
+		if err := editConfigFile(subnetIDstr, networkLower); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func editConfigFile(subnetID string, networkID string) error {
+	warn := "WARNING: This will overwrite your existing config file if there is any, are you sure?"
+	yes, err := prompts.CaptureYesNo(warn)
+	if err != nil {
+		return err
+	}
+	if !yes {
+		ux.Logger.PrintToUser("Canceled by user")
+		return nil
+	}
+	fileBytes, err := os.ReadFile(avagoConfigPath)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("failed to load avalanchego config file %s: %w", avagoConfigPath, err)
+	}
+	if fileBytes == nil {
+		fileBytes = []byte("{}")
+	}
+	var avagoConfig map[string]interface{}
+	if err := json.Unmarshal(fileBytes, &avagoConfig); err != nil {
+		return fmt.Errorf("failed to unpack the config file %s to JSON: %w", avagoConfigPath, err)
+	}
+	avagoConfig["whitelisted-subnets"] = subnetID
```

Comment:
I think we want these to be arrays right? Shouldn't we check for existing values and append so we don't overwrite what's there?

Reply:
Makes sense, but we are only adding to existing subnet IDs _in the config file_ correct? NOT by checking the sidecars etc.

Reply:
Yep. I'm just thinking of the case where I'm validating one subnet already and I want to validate a second. Those values need to be merged, the second shouldn't overwrite the first.